booking/src/main/java/ru/org.dsr/search/SearchKinopoiskJSOUP.java

```diff
@@ -0,0 +1,192 @@
+package ru.org.dsr.search;
+
+import org.apache.log4j.Logger;
+import org.jsoup.nodes.Document;
+import org.jsoup.nodes.Element;
+import org.jsoup.select.Elements;
+import ru.org.dsr.domain.Comment;
+import ru.org.dsr.domain.Item;
+import ru.org.dsr.domain.ItemID;
+import ru.org.dsr.exception.LoadedEmptyBlocksException;
+import ru.org.dsr.exception.NoFoundElementsException;
+import ru.org.dsr.exception.RequestException;
+import ru.org.dsr.exception.RobotException;
+import ru.org.dsr.search.factory.TypeResource;
+
+import java.util.*;
+
+public class SearchKinopoiskJSOUP extends AbstractSearch{
+    private static final Logger log = Logger.getLogger(SearchKinopoiskJSOUP.class);
+
+    private final String FORM_URL_PAGE_COMMENTS = "https://www.kinopoisk.ru%s/reviews/ord/rating/status/all/perpage/200/page/%d";
+    private final String URL_MAIN_ITEM;
+    private Queue<String> movies;
+    private ItemID itemID;
+
+    public SearchKinopoiskJSOUP(ItemID itemID) throws RobotException, RequestException {
+        super("https://www.kinopoisk.ru/index.php?kp_query=", "https://www.kinopoisk.ru");
+        String url = buildUrlSearch(itemID);
+        Document document = getDoc(url);
+        movies = getUrlsMovies(document);
+        URL_MAIN_ITEM = SITE + movies.peek();
+        if (URL_MAIN_ITEM == null)
+            throw new RobotException(document.toString());
+        this.itemID = itemID;
+    }
+
+    SearchKinopoiskJSOUP() {
+        super("https://www.kinopoisk.ru/index.php?kp_query=", "https://www.kinopoisk.ru");
+        URL_MAIN_ITEM = null;
+    }
+
+    @Override
+    public Item getItem() throws RequestException, RobotException {
+        return isEmpty() ? null : initBook();
+    }
+
+    @Override
+    public List<Comment> loadComments(int count) throws RobotException, RequestException {
+        LinkedList<Comment> comments = new LinkedList<>();
+        LinkedList<Comment> currentComments = new LinkedList<>();
+        for(;;) {
+            if (isEmpty()) break;
+            currentComments.addAll(getComments(String.format(FORM_URL_PAGE_COMMENTS, movies.poll(), 1)));
+            Iterator<Comment> it = currentComments.iterator();
+            for (int i = 0; i < count && it.hasNext(); i++) {
+                comments.add(it.next());
+            }
+            if ((count -= currentComments.size()) < 0) break;
+            currentComments.clear();
```

Comment:
Need to rewrite this. I don't understand what is going on. Try to implement things as simple as possible.

Reply:
break; and continue; are dirty hacks. Better to avoid them everywhere and use only in emergency cases.